In [1]:
import numpy as np
import math as mat

# We start with the uncertainties in the lower bound and faint end slope

* This is done in one of two ways:
<ol>
<li> Pick a center value. Find sum of squared distances from other values to that center value. Take the square root and then divide by the number of other values. </li>
<li> Take all values. Find mean, and find sum of squared distances of all values from that mean. Take square root and divide by N-1 </li>
</ol>

In [34]:
#these 6 values are for the final M/L values with alpha = -1.3 and -1.4, with lower bounds of 10^-4 and 10^-3
#all other parameters are held fixed 

#NOTE: these values are for small radii. Will test the large radii values to see if they give a similar uncertainty

values_SR = [217.3,270.3,350.9,198.6,233.4,279.6]
values_LSS = [147.7,182.7,236.0,135.3,158.1,188.4]

In [35]:
# we start with method 1. We pick the center value as index 1, which was calculated with a slope of -1.3 
#and a bound of 10^-4 

def rms_center(center_ind,values):
    rms = 0.0
    for val in values:
        dist = val - values[center_ind]
        rms += dist**2

    rms = (rms**0.5)/(len(values)-1)
    print('The uncertainty is {act:.1f}, or {rel:.2f}% of the center value'\
      .format(act = rms,rel=100*rms/values[1]))
    
    return rms

uncert = rms_center(1,values_SR)
uncert_LSS = rms_center(1,values_LSS)

The uncertainty is 25.2, or 9.33% of the center value
The uncertainty is 16.7, or 9.13% of the center value


In [37]:
# here's method 2. This just simply uses the mean.
def rms_mean(values):
    rms_2 = 0.0
    mean = np.mean(values)
    #print(mean)

    for val in values:
        dist_sqrd = (val - mean)**2
        rms_2 += dist_sqrd

    rms_2 = (rms_2**0.5)/(len(values)-1)

    print('The uncertainty is {act:.1f}, or {rel:.2f}% of the mean'\
          .format(act = rms_2,rel=100*rms_2/mean))
    
    return rms_2

uncert_mean = rms_mean(values_SR)
uncert_mean_LSS = rms_mean(values_LSS)

print('\n' + str(uncert_mean / values[1]))

The uncertainty is 24.5, or 9.49% of the mean
The uncertainty is 16.2, or 9.28% of the mean

0.09073446347972726


+ As we can see here, both methods yeild a fairly similar uncertainty.
<ol>
    <li> Method 1: 25.2 M<sub>sol</sub> L<sub>sol</sub><sup>-1</sup>, 25.2 / 270.3 = 9.33% </li>
    <li> Method 2: 24.5 M<sub>sol</sub> L<sub>sol</sub><sup>-1</sup>, 24.5 / 258.35 = 9.49% </li>
</ol>

# Now, we're going to find the uncertainty in the Leuthaud slope. 
This uses the same functions as above

In [38]:
# the arrays are for values as follows: first value is for a M/L proportional to L^(-0.58), then our center value of 
# L^(-0.54), then L^(-0.50)

leuth_vals_SR = [314.4,270.3,234.9]
leuth_vals_LSS = [212.0,182.7,159.0]

print('Leuthaud Slope, Small Radii')
print('------------------------------ \n')

uncert_leuth_SR = rms_mean(leuth_vals_SR)

print(28.2/270.3)

print('\n Leuthaud Slope, Large Radii')
print('------------------------------ \n')

uncert_leuth_LSS = rms_mean(leuth_vals_LSS)

Leuthaud Slope, Small Radii
------------------------------ 

The uncertainty is 28.2, or 10.31% of the mean
0.10432852386237514

 Leuthaud Slope, Large Radii
------------------------------ 

The uncertainty is 18.8, or 10.17% of the mean
